### PR TITLE
improve: add mobsfscan to CI pipeline (SDKCF-4896)

### PR DESCRIPTION
**Pull request by mdusmangani-jp**
# Description
Add mobsfscan to CI pipeline to find the insecure code patterns.
Ref: https://github.com/MobSF/mobsfscan

## Links
SDKCF-4896

# Checklist
- [ ] I have read the [contributing guidelines](../CONTRIBUTING.md).
- [ ] I wrote/updated tests for new/changed code
- [ ] I removed all sensitive data **before every commit**, including API endpoints and keys
- [ ] I checked that all possible internal and platform/library exceptions are gracefully handled and **will not** crash the host app
- [ ] I checked for open & known issues (especially crashes) for any newly added platform APIs or libraries
- [ ] I tested non-trivial changes on a real device
- [ ] I ran `./gradlew check` without errors

### CODE REVIEW

File: inappmessaging/src/test/java/com/rakuten/tech/mobile/inappmessaging/runtime/view/BaseViewSpec.kt [hunk 1]
```diff
@@ -262,7 +262,7 @@ class BaseViewSpec : BaseTest() {
         if (isValid) {
             imageView?.visibility shouldBeEqualTo View.VISIBLE
         } else {
-            imageView?.visibility shouldBeEqualTo View.GONE
+            imageView?.visibility shouldBeEqualTo  8 // View.GONE
```
Comment (donnie-jp) [inline inappmessaging/src/test/java/com/rakuten/tech/mobile/inappmessaging/runtime/view/BaseViewSpec.kt:265]: Why change a constant var to a hard-coded number here?

Comment (mdusmangani-jp) [reply inappmessaging/src/test/java/com/rakuten/tech/mobile/inappmessaging/runtime/view/BaseViewSpec.kt:265]: Mobscan caught an error for rule Id **android_kotlin_hiddenui** with following description:
"Hidden elements in view can be used to hide data from user. But this data can be leaked"

File: .mobsf [hunk 2]
```diff
@@ -0,0 +1,18 @@
+---
+  ignore-paths:
+  - __MACOSX
+  - skip_dir
+
+  ignore-rules:
```
Comment (vincentdelmundo) [inline .mobsf:6]: What is the reason for these ignored rules?

File: inappmessaging/src/test/java/com/rakuten/tech/mobile/inappmessaging/runtime/view/BaseViewSpec.kt (hunk 1 above)
Comment (vincentdelmundo) [reply inappmessaging/src/test/java/com/rakuten/tech/mobile/inappmessaging/runtime/view/BaseViewSpec.kt:265]: Can this be dismissed as `Used int tests`?

**Review by vincentdelmundo — CHANGES_REQUESTED** [review]
Looks good. Just a few confirmation items.
Will the PR fail (not allowed to merge) if mobsfscan fails?

File: .mobsf (hunk 2 above)
Comment (mdusmangani-jp) [reply .mobsf:6]: According to the doc they evaluate these rules differently. When they do not found any pattern in the entire code then triggered these rules. Please check this doc for details https://github.com/MobSF/mobsfscan/blob/main/mobsfscan/rules/patterns/android/kotlin/best_practices.yaml

File: inappmessaging/src/test/java/com/rakuten/tech/mobile/inappmessaging/runtime/view/BaseViewSpec.kt (hunk 1 above)
Comment (mdusmangani-jp) [reply inappmessaging/src/test/java/com/rakuten/tech/mobile/inappmessaging/runtime/view/BaseViewSpec.kt]: Yes, it is possible

**Review by vincentdelmundo — APPROVED** [review]

**Review by leila-gu — APPROVED** [review]
LGTM

**Review by donnie-jp — APPROVED** [review]

**Pull request merged**